docs/howto/smart_contract_upload.md

```diff
@@ -0,0 +1,138 @@
+        --service scabbard-service-alpha::alpha-node-000,beta-node-000
+        --service scabbard-service-beta::beta-node-000,alpha-node-000
+        --service-peer-group beta-node-000,alpha-node-000 \
+        --service-arg *::admin_keys=<alpha_node_pub_key> \
+        --service-arg scabbard-service-beta::peer_services=alpha-node-000 \
+        --service-arg scabbard-service-alpha::peer_services=beta-node-000 \
+        --service-type *::scabbard \
+        --management tutorial \
+        --auth-type trust
+    ```
+    
+    The `--service-arg`s supplied are specific to the scabbard service and are
+    required in order to use and configue smart contracts. `admin_keys`
+    identify who is allowed to add permissions to a contract and `peer_services`
+    is for identifying which parties are needed for consensus.
+    
+
+    b. Vote to accept the circuit on node beta 
+    
+    ```
+    $ splinter circuit vote \
+        --key <path_to_beta_private_key> \
+        --url http://splinterd-beta:8085 my-circuit --accept
+    ```
+
+    You have now established a circuit between node alpha and node beta based
+    on the provided circuit definition
+
+    c. Verify the circuit has been established 
+
+    ```
+    $ splinter circuit list --url http://splinterd-alpha:8085
+
+    ID            MANAGEMENT MEMBERS
+    <circuit_id>  tutorial   alpha-node-000;beta-node-000
+    ```
+
+2. Package smart contract
+
+    a. Create manifest.yaml
+        
+    The manifest contains the inputs, outputs, version, and the contract
+    name. Below is a sample manifest
+
+    ```
+    name: my_contract
+    version: '1.0'
+    inputs:
+      - '5b7349'
+    outputs:
+      - '5b7349'
+    ```
+
+    b. Bundle the manifest and the compiled wasm for the smart contract into a
+       tarball that has the scar extension 
+
+    ```
+    $ tar -jcvf my_contract.scar my_contract.wasm manifest.yaml 
+    ```
+
+3. Create contract registry for smart contract
+
+    Each contract needs a contract registry to be created that keeps track of
+    versions of the smart contract as well as the contract's owners.
+
+    ```
+    $ scabbard cr create my_contract \
+        --owner <alpha_node_public_key> \
+        --key <path_to_alpha_node_private_key> \
+        --url http://splinterd-alpha:8085 \
+        --service-id my-circuit::scabbard-service-a
+    ```
+
+4. Upload the smart contract.
+
+   ```
+   $ scabbard contract upload ./my_contract.scar \
+       --key <path_to_alpha_node_private_key> \
+       --url http://splinterd-alpha:8085 \
+       --service-id my-circuit::scabbard-service-a
+   ```
+
+5. Create the namespace registry for the smart contract.
+
+    The namespace registry stores the namespace, the owners of the namespace,
+    and the permissions given to that namespace. A namespace is a state address
+    prefix used to identify a portion of state.
+
+   ```
+   $ scabbard ns create <namespace> \
```

Comment:
Can this create a namespace for 5b7349 since it is mentioned in the manifest?

Reply:
Also maybe add a comment that this will need to happen for all namespaces in the manifest